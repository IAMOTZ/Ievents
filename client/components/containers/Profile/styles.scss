@import "../../../styles/variables";

#profile-container {
  #profile-details {
    a {
      text-decoration: none;
      color: black;
      &:hover {
        color: black;
        text-decoration: underline;
      }
    }
    .badge {
      background-color: $custom-blue;
    }
  }
  .text-grey {
    color: grey;
  }
}