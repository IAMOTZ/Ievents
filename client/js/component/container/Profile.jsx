/* global $ */
/* eslint-disable react/prop-types */
import React from 'react';
import { Redirect, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import {
  validateChangePasswordInputs, validateDeleteAccountInputs,
} from '../../helpers/inputValidators';
// Actions.
import {
  changePassword, deleteUser, clearStatus, clearUser,
} from '../../actions/authAction';
import { getAllEvents } from '../../actions/eventActions';
// Common components.
import UserSideNav from '../common/SideNavigation';
import { UserTopNav } from '../common/TopNavigation';
import Header from '../common/Header';
import { LoadingIcon } from '../common/LoadingAnimation';
import { BigAlert, SmallAlert } from '../common/Alert';
// Helpers
import { countCollection } from '../../helpers/helpers';

@connect(store => (
  {
    user: store.user.user,
    authenticated: store.user.status.fetched,
    isAdmin: (store.user.user.role === 'admin'),
    isSuperAdmin: (store.user.user.role === 'superAdmin'),
    events: store.events.events,
    status: {
      changingPassword: store.user.status.changingPassword,
      changingPasswordResolved: store.user.status.changingPasswordResolved,
      changingPasswordRejected: store.user.status.changingPasswordRejected,
      deletingUser: store.user.status.deletingUser,
      deletingUserResolved: store.user.status.deletingUserResolved,
      deletingUserRejected: store.user.status.deletingUserRejected,
    },
  }
))
class Profile extends React.Component {
  constructor() {
    super();
    this.state = {
      formerPassword: '',
      newPassword: '',
      confirmNewPassword: '',
      password: '',
      inputErrors: {
        passwordError: null,
        formerPasswordError: null,
        newPasswordError: null,
        confirmNewPasswordError: null,
      },
    };
  }

  componentDidMount() {
    this.props.dispatch(getAllEvents(this.props.user.token));
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.status.deletingUserResolved) {
      this.props.dispatch(clearUser());
      $('#delete-account-modal').modal('hide');
    }
  }

  componentWillUnmount() {
    this.props.dispatch(clearStatus('CHANGING_PASSWORD'));
    this.props.dispatch(clearStatus('DELETING_USER'));
    $('#delete-account-modal').modal('hide');
    $('#change-password-modal').modal('hide');
  }

  /**
   * Gets user inputs and update the component's state with it.
   * @param {Event} e The event object.
   */
  getInput = (e) => {
    const state = { ...this.state };
    state[e.target.name] = e.target.value;
    this.setState(state);
  }

  /**
   * It clears the inputs of the forms in the modals.
   */
  clearInputs = () => {
    const state = { ...this.state };
    state.formerPassword = '';
    state.newPassword = '';
    state.confirmNewPassword = '';
    state.password = '';
    this.setState(state);
  }

  /**
   * Clears all the inputErrors in the state.
   */
  clearInputErrors = () => {
    const state = { ...this.state };
    state.inputErrors = {
      passwordError: null,
      formerPasswordError: null,
      newPasswordError: null,
      confirmNewPasswordError: null,
    };
    this.setState(state);
  }

  /**
   * It clears all the status in the state relating to changing password.
   */
  clearStateStatus = () => {
    this.props.dispatch(clearStatus('CHANGING_PASSWORD'));
    this.props.dispatch(clearStatus('DELETING_USER'));
  }

  /**
   * Perform some actions when the modal is closed.
   */
  closeModalEffects = () => {
    this.clearInputs();
    this.clearStateStatus();
  }

  /**
   * It dispatches an action to change password.
   */
  changePassword = () => {
    this.props.dispatch(clearStatus('ERROR'));
    const {
      formerPassword, newPassword, confirmNewPassword,
    } = this.state;
    const passwordDetials = {
      formerPassword, newPassword, confirmNewPassword,
    };
    const inputErrors = validateChangePasswordInputs(passwordDetials);
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      this.clearInputErrors();
      const userToken = this.props.user.token;
      this.props.dispatch(changePassword(passwordDetials, userToken));
    }
  }

  /**
   * It dispatches an action to delete the user's account.
   */
  deleteAccount = () => {
    this.props.dispatch(clearStatus('ERROR'));
    const { password } = this.state;
    const inputErrors = validateDeleteAccountInputs({ password });
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      this.clearInputErrors();
      const userToken = this.props.user.token;
      this.props.dispatch(deleteUser(password, userToken));
    }
  }

  render() {
    let component;
    if (!this.props.authenticated) {
      component = <Redirect to="/users/login" />;
    } else {
      component = (
        <div id="profile-container">

          {/* Top Navigation on small and medium screens */}
          <UserTopNav
            name={this.props.user.name}
            title="Profile"
            isAdmin={this.props.isAdmin}
            isSuperAdmin={this.props.isSuperAdmin}
            dispatch={this.props.dispatch}
          />

          <div className="container-fluid">
            <div className="row">

              {/* Side navigation on large sreen */}
              <UserSideNav
                name={this.props.user.name}
                isAdmin={this.props.isAdmin}
                isSuperAdmin={this.props.isSuperAdmin}
                dispatch={this.props.dispatch}
              />

              {/* <!-- Main content --> */}
              <div className="col-lg-10 offset-lg-2">

                {/* Content Header(navigation) on large screen */}
                <Header text="Profile" />

                {/* <!-- Profile Detials --> */}
                <div className="d-flex align-items-center justify-content-center" style={{ minHeight: '500px', marginTop: '70px' }} id="content">
                  <div className="bg-white p-3" style={{ width: '500px' }}>
                    <p className="h1 text-center text-capitalize" style={{ fontWeight: 'lighter', fontSize: '3rem' }}>{this.props.user.name}</p>
                    <p className="h3 mt-3" style={{ fontWeight: 'normal' }}>Events</p>
                    <ul className="list-group">
                      <Link to='/events' className="just-link">
                        <li className="list-group-item d-flex justify-content-between align-items-center text-grey">
                          Done
                          <span className="badge badge-primary badge-pill">
                            {countCollection(this.props.events, event => event.status === 'done')}
                          </span>
                        </li>
                      </Link>
                      <Link to='/events' className="just-link">
                        <li className="list-group-item d-flex justify-content-between align-items-center text-grey">
                          Pending
                          <span className="badge badge-primary badge-pill">
                            {countCollection(this.props.events, event => event.status === 'allowed')}
                          </span>
                        </li>
                      </Link>
                      <Link to='/events' className="just-link">
                        <li className="list-group-item d-flex justify-content-between align-items-center text-grey">
                          Canceled
                          <span className="badge badge-primary badge-pill">
                            {countCollection(this.props.events, event => event.status === 'canceled')}
                          </span>
                        </li>
                      </Link>
                    </ul>
                    <p className="h3 mt-3" style={{ fontWeight: 'normal' }}>Account Settings</p>
                    <ul className="list-group">
                      <a href="" className="just-link" data-toggle="modal" data-target="#change-password-modal">
                        <li className="list-group-item">Change Password</li>
                      </a>
                      <a href="" className="just-link" data-toggle="modal" data-target="#delete-account-modal">
                        <li className="list-group-item">Delete Account</li>
                      </a>
                    </ul>
                  </div>
                </div>
                {/* <!-- /.Profile Details --> */}

                {/* <!-- Change Password Modal --> */}
                <div className="modal fade" id="change-password-modal" tabIndex="-1" role="dialog" data-backdrop="static">
                  <div className="modal-dialog" role="document">
                    <div className="modal-content">
                      <div className="modal-header">
                        <h5 className="modal-title">Change Password</h5>
                        <button
                          type="button"
                          className="close"
                          data-dismiss="modal"
                          aria-label="Close"
                          onClick={this.closeModalEffects}
                          disabled={this.props.status.changingPassword}
                        ><span aria-hidden="true">&times;</span>
                        </button>
                      </div>
                      <div className="modal-body">
                        <LoadingIcon start={this.props.status.changingPassword} size={3} />
                        <BigAlert message={this.props.status.changingPasswordRejected.message} />
                        <BigAlert
                          message={this.props.status.changingPasswordResolved ? 'Password Changed!!' : null}
                          type="success"
                        />
                        <form className="mt-0">
                          <div className="form-group">
                            <label htmlFor="former-password">Former Password</label>
                            <input
                              type="password"
                              className="form-control"
                              id="former-password"
                              name="formerPassword"
                              value={this.state.formerPassword}
                              onChange={this.getInput}
                            />
                            <SmallAlert message={this.state.inputErrors.formerPasswordError} />
                          </div>
                          <div className="form-group">
                            <label htmlFor="new-password">New Password</label>
                            <input
                              type="password"
                              className="form-control"
                              id="new-password"
                              name="newPassword"
                              value={this.state.newPassword}
                              onChange={this.getInput}
                            />
                            <SmallAlert message={this.state.inputErrors.newPasswordError} />
                          </div>
                          <div className="form-group">
                            <label htmlFor="confirm-new-password">Confirm New Password</label>
                            <input
                              type="password"
                              className="form-control"
                              id="confirm-new-password"
                              name="confirmNewPassword"
                              value={this.state.confirmNewPassword}
                              onChange={this.getInput}
                            />
                            <SmallAlert message={this.state.inputErrors.confirmNewPasswordError} />
                          </div>
                        </form>
                      </div>
                      <div className="modal-footer">
                        <button
                          type="button"
                          className="btn btn-dark pointer-button"
                          data-dismiss="modal"
                          onClick={this.closeModalEffects}
                          disabled={this.props.status.changingPassword}
                        >Cancel
                        </button>
                        <button
                          type="button"
                          className="btn btn-primary pointer-button"
                          onClick={this.changePassword}
                          disabled={this.props.status.changingPassword}
                        >Change
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
                {/* <!-- /.Change Password Modal --> */}

                {/* <!-- Delete Account Modal --> */}
                <div className="modal fade" id="delete-account-modal" tabIndex="-1" role="dialog" data-backdrop="static">
                  <div className="modal-dialog" role="document">
                    <div className="modal-content">
                      <div className="modal-header">
                        <h5 className="modal-title">Delete Account</h5>
                        <button
                          type="button"
                          className="close"
                          data-dismiss="modal"
                          aria-label="Close"
                          onClick={this.closeModalEffects}
                          disabled={this.props.status.deletingUser}
                        ><span aria-hidden="true">&times;</span>
                        </button>
                      </div>
                      <div className="modal-body">
                        <p className="h6">Are you sure you wan't to delete your account? </p>
                        <span className="text-grey">This action is irreversible, you would loose all your data on this platform.</span>
                        <div className="form-group mt-3">
                          <label htmlFor="password" className="h6">Your Password</label>
                          <input
                            type="password"
                            className="form-control"
                            id="password"
                            placeholder="Input you password to continue"
                            name="password"
                            value={this.state.password}
                            onChange={this.getInput}
                          />
                          <SmallAlert message={this.state.inputErrors.passwordError} />
                          <LoadingIcon start={this.props.status.deletingUser} size={3} />
                          <br />
                          <BigAlert message={this.props.status.deletingUserRejected.message} />
                        </div>
                      </div>
                      <div className="modal-footer">
                        <button
                          type="button"
                          className="btn btn-dark pointer-button"
                          data-dismiss="modal"
                          onClick={this.closeModalEffects}
                          disabled={this.props.status.deletingUser}
                        >Cancel
                        </button>
                        <button
                          type="button"
                          className="btn btn-danger pointer-button"
                          onClick={this.deleteAccount}
                          disabled={this.props.status.deletingUser}
                        >Delete
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
                {/* <!-- /.Delete Account Modal --> */}

              </div>
            </div>
          </div>
        </div>
      );
    }
    return component;
  }
}

export default Profile;
