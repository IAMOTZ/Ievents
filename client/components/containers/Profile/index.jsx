/* global $ */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  validateChangePasswordInputs, validateDeleteAccountInputs,
} from '../../../helpers/inputValidators';
import { logOut } from '../../../actions/authAction';
import { changePassword, deleteAccount } from '../../../actions/userActions';
import { getAllEvents } from '../../../actions/eventActions';
import { stopAsyncProcess } from '../../../actions/commonActions';
import * as asyncProcess from '../../../actions/asyncProcess';
import './styles.scss';
import View from './View';

@connect((store) => {
  const { user } = store.authReducer;
  return {
    userName: user.name,
    userToken: user.token,
    isAdmin: user.role === 'admin' || user.role === 'superAdmin',
    isSuperAdmin: user.role === 'superAdmin',
    events: store.fetchEventsReducer.events,
    changingPasswordStarted: store.changePasswordReducer.changingPasswordStarted,
    changingPasswordResolved: store.changePasswordReducer.changingPasswordResolved,
    changingPasswordError: store.changePasswordReducer.changingPasswordError,
    deletingAccountStarted: store.deleteAccountReducer.deletingAccountStarted,
    deletingAccountResolved: store.deleteAccountReducer.deletingAccountResolved,
    deletingAccountError: store.deleteAccountReducer.deletingAccountError,
  };
})
class Profile extends React.Component {
  constructor() {
    super();
    this.state = {
      formerPassword: '',
      newPassword: '',
      confirmNewPassword: '',
      password: '',
      inputErrors: {
        passwordError: null,
        formerPasswordError: null,
        newPasswordError: null,
        confirmNewPasswordError: null,
      },
    };
  }

  componentDidMount() {
    this.props.dispatch(getAllEvents(this.props.userToken));
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.deletingAccountResolved) {
      this.props.dispatch(logOut());
      $('#delete-account-modal').modal('hide');
    }
  }

  componentWillUnmount() {
    this.props.dispatch(stopAsyncProcess(asyncProcess.CHANGING_PASSWORD));
    this.props.dispatch(stopAsyncProcess(asyncProcess.DELETING_ACCOUNT));
    $('#delete-account-modal').modal('hide');
    $('#change-password-modal').modal('hide');
  }

  /**
   * Stores the user inputs in the state of this component.
   * @param {Event} event The event object.
   */
  getInput = (event) => {
    const state = { ...this.state };
    state[event.target.name] = event.target.value;
    this.setState(state);
  }

  /**
   * It clears the inputs of the forms in the modals.
   */
  clearInputs = () => {
    const state = { ...this.state };
    state.formerPassword = '';
    state.newPassword = '';
    state.confirmNewPassword = '';
    state.password = '';
    this.setState(state);
  }

  /**
   * Clears all the inputErrors in the state.
   */
  clearInputErrors = () => {
    const state = { ...this.state };
    state.inputErrors = {
      passwordError: null,
      formerPasswordError: null,
      newPasswordError: null,
      confirmNewPasswordError: null,
    };
    this.setState(state);
  }

  /**
   * It stops the asynchronous process that might have being started.
   */
  killAsyncProcess = () => {
    this.props.dispatch(stopAsyncProcess(asyncProcess.CHANGING_PASSWORD));
    this.props.dispatch(stopAsyncProcess(asyncProcess.DELETING_ACCOUNT));
  }

  /**
   * Perform some clean up actions when the modal is closed.
   */
  closeModalEffects = () => {
    this.clearInputs();
    setTimeout(() => {
      this.clearInputErrors();
    }, 1000);
    this.killAsyncProcess();
  }

  /**
   * It dispatches an action to change password.
   */
  changePassword = () => {
    this.props.dispatch(stopAsyncProcess(asyncProcess.CHANGING_PASSWORD));
    const {
      formerPassword, newPassword, confirmNewPassword,
    } = this.state;
    const passwordDetials = {
      formerPassword, newPassword, confirmNewPassword,
    };
    const inputErrors = validateChangePasswordInputs(passwordDetials);
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      this.clearInputErrors();
      const { userToken } = this.props;
      this.props.dispatch(changePassword(passwordDetials, userToken));
    }
  }

  /**
   * It dispatches an action to delete the user's account.
   */
  deleteAccount = () => {
    this.props.dispatch(stopAsyncProcess(asyncProcess.DELETING_ACCOUNT));
    const { password } = this.state;
    const inputErrors = validateDeleteAccountInputs({ password });
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      this.clearInputErrors();
      const { userToken } = this.props;
      this.props.dispatch(deleteAccount(password, userToken));
    }
  }

  render() {
    return (
      <View
        userName={this.props.userName}
        events={this.props.events}
        isAdmin={this.props.isAdmin}
        isSuperAdmin={this.props.isSuperAdmin}
        dispatch={this.props.dispatch}
        changingPasswordStarted={this.props.changingPasswordStarted}
        changingPasswordError={this.props.changingPasswordError}
        changingPasswordResolved={this.props.changingPasswordResolved}
        deletingAccountStarted={this.props.deletingAccountStarted}
        deletingAccountError={this.props.deletingAccountError}
        formerPassword={this.state.formerPassword}
        newPassword={this.state.newPassword}
        confirmNewPassword={this.state.confirmNewPassword}
        password={this.state.password}
        inputErrors={this.state.inputErrors}
        getInput={this.getInput}
        closeModalEffects={this.closeModalEffects}
        deleteAccount={this.deleteAccount}
        changePassword={this.changePassword}
      />
    );
  }
}

Profile.defaultProps = {
  userName: '',
  userToken: '',
  isAdmin: false,
  isSuperAdmin: false,
  events: [],
  changingPasswordStarted: false,
  changingPasswordResolved: false,
  changingPasswordError: '',
  deletingAccountStarted: false,
  deletingAccountResolved: false,
  deletingAccountError: '',
  dispatch: () => {},
};

/* eslint-disable react/forbid-prop-types */
Profile.propTypes = {
  userName: PropTypes.string,
  userToken: PropTypes.string,
  isAdmin: PropTypes.bool,
  isSuperAdmin: PropTypes.bool,
  events: PropTypes.array,
  changingPasswordStarted: PropTypes.bool,
  changingPasswordResolved: PropTypes.bool,
  changingPasswordError: PropTypes.string,
  deletingAccountStarted: PropTypes.bool,
  deletingAccountResolved: PropTypes.bool,
  deletingAccountError: PropTypes.string,
  dispatch: PropTypes.func,
};

export default Profile;
