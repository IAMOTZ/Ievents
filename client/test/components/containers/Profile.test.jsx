import React from 'react';
import Profile from '../../../components/containers/Profile';
import ProfileView from '../../../components/containers/Profile/View';
import DeleteAccountModal from '../../../components/containers/Profile/DeleteAccountModal';
import ChangePasswordModal from '../../../components/containers/Profile/ChangePasswordModal';

describe('<Profile />', () => {
  const props = {
    userName: 'tester',
    userToken: 'sampleToken',
    isAdmin: false,
    isSuperAdmin: false,
    events: [],
    changingPasswordStarted: false,
    changingPasswordResolved: false,
    changingPasswordError: null,
    deletingAccountStarted: false,
    deletingAccountResolved: false,
    deletingAccountError: null,
    dispatch: jest.fn(),
  };
  const alterProps = newProps => ({ ...props, ...newProps });
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Profile {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Behaviour:', () => {
    let wrapper; let wrapperInstance;
    beforeEach(() => {
      props.dispatch.mockReset();
      wrapper = shallow(<Profile {...props} />);
      wrapperInstance = wrapper.instance();
    });
    // The names on the following test block refers
    // to the instance method that handles the behaviour tested.
    describe('COMOPONENT_DID_MOUNT', () => {
      it('should dispatch an action', () => {
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });
    describe('COMOPONENT_WILL_RECIEVE_PROPS', () => {
      it('should dispatch an action if deleting account resolved', () => {
        props.dispatch.mockReset();
        wrapper.setProps(alterProps({ deletingAccountResolved: true }));
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });
    describe('COMOPONENT_WILL_UNMOUNT', () => {
      it('should dispatch two actions', () => {
        props.dispatch.mockReset();
        wrapper.unmount();
        expect(props.dispatch.mock.calls.length).toEqual(2);
      });
    });

    describe('GET_INPUT', () => {
      it('should update the state with user input', () => {
        const userInput = 'newPassword123';
        const fakeEvent = { target: { name: 'newPassword', value: userInput } };
        wrapperInstance.getInput(fakeEvent);
        expect(wrapper.state('newPassword')).toEqual(userInput);
      });
    });

    describe('CLEAR_INPUTS', () => {
      it('should clear the user inputs from the state', () => {
        wrapper.setState({ formerPassword: 'myFormerPassword' });
        wrapperInstance.clearInputs();
        expect(wrapper.state('formerPassword')).toEqual('');
      });
    });

    describe('CLEAR_INPUT_ERRORS', () => {
      it('should clear input errors in the state', () => {
        const passwordError = 'Password is required';
        wrapper.setState({ inputErrors: { passwordError } });
        wrapperInstance.clearInputErrors();
        expect(wrapper.state('inputErrors').passwordError).toBeNull();
      });
    });

    describe('KILL_ASYNC_PROCESS', () => {
      it('should dispatch two actions', () => {
        props.dispatch.mockReset();
        wrapperInstance.killAsyncProcess();
        expect(props.dispatch.mock.calls.length).toEqual(2);
      });
    });

    describe('CLOSE_MODAL_EFFECT', () => {
      it('should clear user inputs in the state', () => {
        wrapper.setState({ formerPassword: 'myFormerPassword' });
        wrapperInstance.closeModalEffects();
        expect(wrapper.state('formerPassword')).toEqual('');
      });
      it('should dispatch two actions', () => {
        props.dispatch.mockReset();
        wrapperInstance.closeModalEffects();
        expect(props.dispatch.mock.calls.length).toEqual(2);
      });
    });

    describe('CHANGE_PASSWORD', () => {
      it('should dispatch an action and set inputErros if validation fails', () => {
        const emptyPassword = '';
        wrapper.setState({ formerPassword: emptyPassword });
        props.dispatch.mockReset();
        wrapperInstance.changePassword();
        expect(typeof wrapper.state('inputErrors').formerPasswordError).toEqual('string');
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
      it('should dispatch two acitons and clear inputErrors if validation pass', () => {
        const correctData = {
          formerPassword: 'Formerpassword123',
          newPassword: 'Newpassword123',
          confirmNewPassword: 'Newpassword123',
        };
        const previousError = 'Former password is required';
        wrapper.setState({
          ...correctData, inputErrors: { formerPasswordError: previousError }
        });
        props.dispatch.mockReset();
        wrapperInstance.changePassword();
        expect(wrapper.state('inputErrors').formerPasswordError).toBeNull();
        expect(props.dispatch.mock.calls.length).toEqual(2);
      });
    });

    describe('DELETE_ACCOUNT', () => {
      it('should dispatch an action and set inputErros if validation fails', () => {
        const emptyPassword = '';
        wrapper.setState({ password: emptyPassword });
        props.dispatch.mockReset();
        wrapperInstance.deleteAccount();
        expect(typeof wrapper.state('inputErrors').passwordError).toEqual('string');
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
      it('should dispatch two acitons and clear inputErrors if validation pass', () => {
        const correctData = { password: 'Password123' };
        const previousError = 'Password is required';
        wrapper.setState({
          ...correctData, inputErrors: { passwordError: previousError }
        });
        props.dispatch.mockReset();
        wrapperInstance.deleteAccount();
        expect(wrapper.state('inputErrors').passwordError).toBeNull();
        expect(props.dispatch.mock.calls.length).toEqual(2);
      });
    });
  });
});

describe('<ProfileView />', () => {
  const props = {
    userName: 'tester',
    isAdmin: false,
    isSuperAdmin: false,
    dispatch: jest.fn(),
    events: [],
    changingPasswordStarted: false,
    changingPasswordError: null,
    changingPasswordResolved: false,
    deletingAccountStarted: false,
    deletingAccountError: null,
    formerPassword: '',
    newPassword: '',
    confirmNewPassword: '',
    password: '',
    inputErrors: {},
    getInput: () => { },
    closeModalEffects: () => { },
    deleteAccount: () => { },
    changePassword: () => { },
  };
  describe('Rendering', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<ProfileView {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('<ChangePasswordModal />', () => {
  const props = {
    changingPasswordStarted: false,
    changingPasswordError: null,
    changingPasswordResolved: false,
    formerPassword: '',
    newPassword: '',
    confirmNewPassword: '',
    inputErrors: {},
    getInput: () => { },
    closeModalEffects: () => { },
    changePassword: () => { },
  };
  describe('Rendering', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<ChangePasswordModal {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('<DeleteAccountModal />', () => {
  const props = {
    deletingAccountStarted: false,
    deletingAccountError: null,
    password: null,
    inputErrors: {},
    getInput: () => {},
    closeModalEffects: () => {},
    deleteAccount: () => {},
  };
  describe('Rendering', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<DeleteAccountModal {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

