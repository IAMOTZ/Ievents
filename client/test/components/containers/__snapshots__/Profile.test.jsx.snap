// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePasswordModal /> Rendering should render correctly 1`] = `
<div
  className="modal fade"
  data-backdrop="static"
  id="change-password-modal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Change Password
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <LoadingIcon
          size={3}
          start={false}
        />
        <BigAlert
          message={null}
        />
        <BigAlert
          message={null}
          type="success"
        />
        <form
          className="mt-0"
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="former-password"
            >
              Former Password
            </label>
            <input
              className="form-control"
              id="former-password"
              name="formerPassword"
              onChange={[Function]}
              type="password"
              value=""
            />
            <SmallAlert />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="new-password"
            >
              New Password
            </label>
            <input
              className="form-control"
              id="new-password"
              name="newPassword"
              onChange={[Function]}
              type="password"
              value=""
            />
            <SmallAlert />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="confirm-new-password"
            >
              Confirm New Password
            </label>
            <input
              className="form-control"
              id="confirm-new-password"
              name="confirmNewPassword"
              onChange={[Function]}
              type="password"
              value=""
            />
            <SmallAlert />
          </div>
        </form>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn ie-dark-button"
          data-dismiss="modal"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn ie-blue-button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Change
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DeleteAccountModal /> Rendering should render correctly 1`] = `
<div
  className="modal fade"
  data-backdrop="static"
  id="delete-account-modal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Delete Account
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p
          className="h6"
        >
          Are you sure you wan't to delete your account? 
        </p>
        <span
          className="text-grey"
        >
          This action is irreversible, you would loose all your data on this platform.
        </span>
        <div
          className="form-group mt-3"
        >
          <label
            className="h6"
            htmlFor="password"
          >
            Your Password
          </label>
          <input
            className="form-control"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Input you password to continue"
            type="password"
            value={null}
          />
          <SmallAlert />
          <LoadingIcon
            size={3}
            start={false}
          />
          <br />
          <BigAlert
            message={null}
          />
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-dark pointer-button"
          data-dismiss="modal"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-danger pointer-button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Profile /> Rendering: should render correctly 1`] = `
<View
  changePassword={[Function]}
  changingPasswordError={null}
  changingPasswordResolved={false}
  changingPasswordStarted={false}
  closeModalEffects={[Function]}
  confirmNewPassword=""
  deleteAccount={[Function]}
  deletingAccountError={null}
  deletingAccountStarted={false}
  dispatch={
    [MockFunction] {
      "calls": Array [
        Array [
          [Function],
        ],
      ],
    }
  }
  events={Array []}
  formerPassword=""
  getInput={[Function]}
  inputErrors={
    Object {
      "confirmNewPasswordError": null,
      "formerPasswordError": null,
      "newPasswordError": null,
      "passwordError": null,
    }
  }
  isAdmin={false}
  isSuperAdmin={false}
  newPassword=""
  password=""
  userName="tester"
/>
`;

exports[`<ProfileView /> Rendering should render correctly 1`] = `
<div
  id="profile-container"
>
  <AuthTopNavigation
    dispatch={[MockFunction]}
    isAdmin={false}
    isSuperAdmin={false}
    name="tester"
    title="Profile"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <SideNavigation
        dispatch={[MockFunction]}
        isAdmin={false}
        isSuperAdmin={false}
        name="tester"
      />
      <div
        className="col-lg-10 offset-lg-2"
      >
        <Header
          text="Profile"
        />
        <div
          className="d-flex align-items-center justify-content-center"
          id="profile-details"
          style={
            Object {
              "marginTop": "70px",
              "minHeight": "500px",
            }
          }
        >
          <div
            className="bg-white p-3"
            style={
              Object {
                "width": "500px",
              }
            }
          >
            <p
              className="h1 text-center text-capitalize"
              style={
                Object {
                  "fontSize": "3rem",
                  "fontWeight": "lighter",
                }
              }
            >
              tester
            </p>
            <p
              className="h3 mt-3"
              style={
                Object {
                  "fontWeight": "normal",
                }
              }
            >
              Events
            </p>
            <ul
              className="list-group"
            >
              <li
                className="list-group-item d-flex justify-content-between align-items-center text-grey"
              >
                Done
                <span
                  className="badge badge-primary badge-pill"
                >
                  0
                </span>
              </li>
              <li
                className="list-group-item d-flex justify-content-between align-items-center text-grey"
              >
                Allowed
                <span
                  className="badge badge-primary badge-pill"
                >
                  0
                </span>
              </li>
              <li
                className="list-group-item d-flex justify-content-between align-items-center text-grey"
              >
                Canceled
                <span
                  className="badge badge-primary badge-pill"
                >
                  0
                </span>
              </li>
            </ul>
            <p
              className="h3 mt-3"
              style={
                Object {
                  "fontWeight": "normal",
                }
              }
            >
              Account Settings
            </p>
            <ul
              className="list-group account-settings"
            >
              <a
                className="just-link no-outline-btn"
                data-toggle="modal"
                href="#change-password-modal"
              >
                <li
                  className="list-group-item"
                >
                  Change Password
                </li>
              </a>
              <a
                className="just-link no-outline-btn"
                data-toggle="modal"
                href="#delete-account-modal"
              >
                <li
                  className="list-group-item"
                >
                  Delete Account
                </li>
              </a>
            </ul>
          </div>
        </div>
        <ChangePasswordModal
          changePassword={[Function]}
          changingPasswordError={null}
          changingPasswordResolved={false}
          changingPasswordStarted={false}
          closeModalEffects={[Function]}
          confirmNewPassword=""
          formerPassword=""
          getInput={[Function]}
          inputErrors={Object {}}
          newPassword=""
        />
        <DeleteAccountModal
          closeModalEffects={[Function]}
          deleteAccount={[Function]}
          deletingAccountError={null}
          deletingAccountStarted={false}
          getInput={[Function]}
          inputErrors={Object {}}
          password=""
        />
      </div>
    </div>
  </div>
</div>
`;
